### PR TITLE
feat: add compression option to MongoClientOptions

### PR DESCRIPTION
**Pull request by jtassin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/mongodb/node-mongodb-native/blob/e90af9610eb25ab895a8e91d5b373ff87d92c5b3/test/disabled/server.test.js#L127>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -360,6 +360,12 @@ export interface MongoClientOptions extends
      * Mechanism for authentication: DEFAULT, GSSAPI, PLAIN, MONGODB-X509, 'MONGODB-CR', SCRAM-SHA-1 or SCRAM-SHA-256
      */
     authMechanism?: 'DEFAULT' | 'GSSAPI' | 'PLAIN' | 'MONGODB-X509' | 'MONGODB-CR' | 'SCRAM-SHA-1' | 'SCRAM-SHA-256' | string;
+
+    /** Type of compression to use */
+    compression?: {
+        /** The selected compressors in preference order */
+        compressors?: Array<('zstd' | 'snappy' | 'zlib')>;
```
Comment (CaselIT) [inline types/mongodb/index.d.ts:367]: Is there any documentation regarding `'zstd'`.
http://mongodb.github.io/node-mongodb-native/3.6/api/MongoClient.html#.connect list only the other two

Comment (jtassin) [reply types/mongodb/index.d.ts:367]: @CaselIT yes in the global mongodb doc : https://docs.mongodb.com/manual/core/wiredtiger/#compression

It's a high compression / low CPU mode added in mongo 4.2.

The node-mongodb-native documentation is far from up to date :/

Comment (LinusU) [reply types/mongodb/index.d.ts:367]: Are you sure that the Node.js driver supports `zstd` though? 🤔 

Doesn't seem like it?
https://github.com/mongodb/node-mongodb-native/search?q=zstd&unscoped_q=zstd

Comment (jtassin) [reply types/mongodb/index.d.ts:367]: hummmm, you're right.
Well seen, I'll remove it from the list

Comment (jtassin) [reply types/mongodb/index.d.ts]: done

**Review by LinusU — APPROVED** [review]
👍

**Review by CaselIT — APPROVED** [review]

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (CaselIT) [reply types/mongodb/index.d.ts]: maybe an issue could be opened in the nodejs driver repo?

**Comment by jtassin** [comment]
Ready to merge

**Pull request merged**